free/src/main/scala/cats/free/InvariantCoyoneda.scala

```diff
@@ -0,0 +1,108 @@
+package cats
+package free
+
+/**
+ * The free invariant functor on `F`. This is isomorphic to `F` as long as `F` itself is a
+ * invariant functor. The function from `F[A]` to `InvariantCoyoneda[F,A]` exists even when
+ * `F` is not an invariant functor. Implemented using a List of functions for stack-safety.
+ */
+sealed abstract class InvariantCoyoneda[F[_], A] extends Serializable { self =>
+  import InvariantCoyoneda.{unsafeApply, Aux}
+
+  /**
+   * The pivot between `fi` and `k`, usually existential.
+   */
+  type Pivot
+
+  /**
+   * The underlying value.
+   */
+  val fi: F[Pivot]
+
+  /**
+   * The list of transformer functions, to be composed and lifted into `F` by `run`.
+   */
+  private[cats] val ks0: List[Any => Any]
+
+  /**
+   * The list of transformer functions, to be composed and lifted into `F` by `run`.
+   */
+  private[cats] val ks1: List[Any => Any]
+
+  /**
+   * The list of transformer functions composed into a single function, to be lifted into `F` by `run`.
+   */
+  final def k0: Pivot => A = Function.chain(ks0.reverse)(_).asInstanceOf[A]
+
+  /**
+   * The composed transformer function, to be lifted into `F` by `run`.
+   */
+  final def k1: A => Pivot = Function.chain(ks1)(_).asInstanceOf[Pivot]
```

Comment:
Are there better names for these? Also, I see the comments are broken, will fix.

Reply:
I think this is fine, just to keep it consistent with the co/contravariant versions.